packages/domain-picker/src/domain-picker/index.tsx

```diff
@@ -376,6 +381,9 @@ const DomainPicker: FunctionComponent< Props > = ( {
 											times( placeholdersCount, ( i ) => (
 												<SuggestionItemPlaceholder type={ itemType } key={ i } />
 											) ) }
+										{ ! areDependenciesLoading && existingSubdomain && onUseYourDomainClick && (
```

Comment:
I'm not sure about the `areDependenciesLoading` check but we definitely don't want `existingSubdomain` to be a condition for displaying this button.
 
```suggestion
										{ ! areDependenciesLoading && onUseYourDomainClick && (
```

Reply:
@razvanpapadopol you are right. I have made some markup and conditional changes around this code but just to explain why:

`areDependenciesLoading` - When domain suggestions are loading, we don't need to perform this check for "Use a domain I own" because they are not a dependency on this item similar to existing sub-domains.
`existingSubDomain` - Not sure how this snuck in, or how I left it but you're right and it's been removed.

I've also amended the CSS selectors for `border-radius` on the first and last items in [this commit](https://github.com/Automattic/wp-calypso/pull/48119/commits/7dd94e90d227f63e4f187758d9f7501d0253515f) and also wrapped the listed domain items in a container because of the following reasons:

* Placeholders and domain suggestions have different "types" (div/label) so the last placeholder isn't actually "last" because "Use a domain I own" is below it. Changing the selector to `last-child` fixed this.
* The "Show all domains" was in the same list so was also being considered as the `last-child` meaning I had to add a containing element to the domain suggestion list.
* I could have just moved "Show all domains" outside of this element but it would have created a flexbox bug in the scenario where `showDomainCategories` was `true` because the `flex-direciton` must be `row` to accommodate this.

Reply:
Thanks, @tjcafferkey, for the fixes and for providing the relevant context 👍 

> last placeholder isn't actually "last" because "Use a domain I own" is below it. Changing the selector to last-child fixed this.

good catch!

Just one more note, after rebase, please make sure the fixes from https://github.com/Automattic/wp-calypso/pull/48172 merged yesterday are still in place.

Reply:
@razvanpapadopol  Done 👍🏻